environs/bootstrap/bootstrap.go

```diff
@@ -218,8 +206,10 @@ func Bootstrap(ctx environs.BootstrapContext, environ environs.Environ, args Boo
 	// If no arch is specified as a constraint, we'll bootstrap
 	// on the same arch as the client used to bootstrap.
 	var bootstrapArch string
-	if bootstrapConstraints.Arch != nil {
-		bootstrapArch = *bootstrapConstraints.Arch
+	if args.BootstrapConstraints.Arch != nil {
```

Comment:
revert back to checking just BootstrapConstraints, since they're expected to be merged already